pkg/ingress/kube/configmap/tracing.go

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2022 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configmap
+
+import (
+	"fmt"
+	"sync/atomic"
+
+	"github.com/alibaba/higress/pkg/ingress/kube/util"
+	. "github.com/alibaba/higress/pkg/ingress/log"
+	networking "istio.io/api/networking/v1alpha3"
+	"istio.io/istio/pkg/config"
+	"istio.io/istio/pkg/config/schema/gvk"
+	listersv1 "k8s.io/client-go/listers/core/v1"
+	"sigs.k8s.io/yaml"
+)
+
+type TracingMgr struct {
+	Namespace               string
+	HigressConfigController HigressConfigController
+	HigressConfigLister     listersv1.ConfigMapNamespaceLister
+	higressConfig           atomic.Value
+}
+
+func NewTracingMgr(namespace string, higressConfigController HigressConfigController, higressConfigLister listersv1.ConfigMapNamespaceLister) *TracingMgr {
+
+	tracingMgr := &TracingMgr{
+		Namespace:               namespace,
+		HigressConfigController: higressConfigController,
+		HigressConfigLister:     higressConfigLister,
+		higressConfig:           atomic.Value{},
+	}
+	tracingMgr.HigressConfigController.AddEventHandler(tracingMgr.AddOrUpdateHigressConfig)
```

Comment:
Global event handlers should trigger all configuration logic updates，not just tracing.